sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -479,10 +488,10 @@ public boolean advance() throws IOException {
         checkpointMark.sessionIds.add(deliveryTag);
 
         current = new RabbitMqMessage(source.spec.routingKey(), delivery);
-        currentTimestamp = new Instant(delivery.getProperties().getTimestamp());
-        if (currentTimestamp.isBefore(checkpointMark.oldestTimestamp)) {
-          checkpointMark.oldestTimestamp = currentTimestamp;
-        }
+        Date deliveryTimestamp = delivery.getProperties().getTimestamp();
```

Comment:
The meat of the change. Note I'm having trouble finding any documentation on exactly how `getProperties().getTimestamp()` is populated. I'm falling back to `Instant.now()` as a safeguard (which should actually be safe).